### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by MaksNochvai**

### CODE REVIEW

File: app/create_file.py [hunk 1]
```diff
@@ -1 +1,29 @@
...
+    with file_path.open("a") as file:
+        file.write(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S\n"))
+        for count, input_string in enumerate(iter(input, "stop")):
+            file.write(f"{count} {input_string}\n")
+
+
+def create_file() -> None:
+    if "-d" in argv and "-f" in argv:
+        create_path(path.sep.join(
+            argv[argv.index("-d") + 1:argv.index("-f")]), argv[-1]
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:24]: But if -f will be first?
python create_file.py -f file.txt -d dir1 dir2

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/create_file.py [hunk 2]
```diff
@@ -1 +1,32 @@
...
+            file.write(f"{count} {input_string}\n")
+
+
+def create_file() -> None:
+    if "-f" in argv:
+        file_index = argv.index("-f")
+        file_name = argv[file_index + 1]
+        if "-d" in argv:
+            dir_index = argv.index("-d")
+            dir_path = path.sep.join(argv[dir_index + 1:file_index])
```
Comment (ArsenPidhoretskyi) [inline app/create_file.py:27]: It will does not work in next case:
python create_file.py -f file.txt -d dir1 dir2

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/create_file.py [hunk 3]
```diff
@@ -1 +1,37 @@
...
+
+
+def create_file() -> None:
+    if "-f" in argv:
+        file_index = argv.index("-f")
+        file_name = argv[file_index + 1]
+        dir_index = None
+        for i, arg in enumerate(argv):
+            if arg == "-d" and i < file_index:
+                dir_index = i
```
[the remark this answers is not in the document]
Comment (ArsenPidhoretskyi) [reply app/create_file.py:28]: Remove this

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/create_file.py [hunk 4]
```diff
@@ -1 +1,40 @@
...
+
+def create_dir() -> None:
+    input_command = sys.argv
+    file_name = ""
+    parent_dir = ""
+    if ("-f" and "-d") in input_command:
+        index_f, index_d = input_command.index("-f"), input_command.index("-d")
+        file_name = input_command[index_f + 1]
+        if index_d < index_f:
+            parent_dir = "/".join(input_command[(index_d + 1):index_f])
```
Comment (olena-hul) [inline app/create_file.py:14]: Use os.path.join

File: app/create_file.py [hunk 5]
```diff
@@ -1 +1,40 @@
...
+        if index_d > index_f:
+            parent_dir = "/".join(input_command[index_d + 1:])
```
Comment (olena-hul) [inline app/create_file.py:16]: And here

File: app/create_file.py [hunk 6]
```diff
@@ -1 +1,40 @@
...
+    if "-f" not in input_command and "-d" not in input_command:
+        print("Please specify at least one flag -f or -d")
+        sys.exit(1)
+    if "-f" in input_command and "-d" not in input_command:
+        file_name = input_command[1]
+    if "-d" in input_command and "-f" not in input_command:
+        parent_dir = "/".join(input_command[1:])
+    path = os.path.join(parent_dir, file_name)
+    os.makedirs(path)
```
Comment (olena-hul) [inline app/create_file.py:25]: The exception will be risen here in case the specified path exists

File: app/create_file.py [hunk 7]
```diff
@@ -1 +1,40 @@
...
+        if index_d < index_f:
+            parent_dir = "/".join(input_command[(index_d + 1):index_f])
+        if index_d > index_f:
+            parent_dir = "/".join(input_command[index_d + 1:])
+    if "-f" not in input_command and "-d" not in input_command:
+        print("Please specify at least one flag -f or -d")
+        sys.exit(1)
+    if "-f" in input_command and "-d" not in input_command:
+        file_name = input_command[1]
+    if "-d" in input_command and "-f" not in input_command:
```
Comment (olena-hul) [inline app/create_file.py:22]: I think that the second part of checks is redundant. If the both flags exist, the first `if` will be triggered

File: app/create_file.py (hunk 2 above)
Comment (olena-hul) [reply app/create_file.py]: @MaksNochvai Still no screenshots present

**Review by olena-hul — CHANGES_REQUESTED** [review]

**Pull request closed without merging**